src/mock.utils.ts

```diff
@@ -87,21 +92,68 @@ export class MockEthersProvider {
     return this;
   }
 
-  public addFilteredLogs(filter: Filter | FilterByBlockHash, logs: Log[]): MockEthersProvider {
-    const matcher = {
-      ...filter,
-      topics: (filter.topics)
-        ? expect.arrayContaining(filter.topics.map(el => (Array.isArray(el))? expect.arrayContaining(el) : el))
-        : undefined,
-    };
+  public addLogs(...logs: Log[]): MockEthersProvider {
+    this.logs.push(...logs);
+    return this;
+  }
+
+  private _getLogs(filter: Filter | FilterByBlockHash): Log[] {
+    let logs = this.logs;
+
+    if (filter.address) logs = logs.filter((log) => log.address === filter.address);
+
+    if (filter.topics) {
+      const filterTopics = filter.topics!;
+
+      logs = logs.filter((log) => {
+        if (filterTopics.length !== log.topics.length) {
+          return false;
+        }
+
+        for (let i = 0; i < filterTopics.length; i++) {
+          const logTopic = log.topics[i];
+          const filterTopic = filterTopics[i];
 
-    when(this.getLogs).calledWith(expect.objectContaining(matcher)).mockReturnValue(logs);
+          if (filterTopic === null) continue;
+          if (!logTopic) return false;
 
+          if (Array.isArray(filterTopic)) {
+            if (!filterTopic.includes(logTopic)) {
+              return false;
+            }
+          } else if (logTopic !== filterTopic) {
+            return false;
+          }
+        }
+
+        return true;
+      });
+    }
+
+    if ("fromBlock" in filter && filter.fromBlock) {
+      logs = logs.filter((log) => log.blockNumber >= filter.fromBlock!);
+    }
+
+    if ("toBlock" in filter && filter.toBlock) {
+      logs = logs.filter((log) => log.blockNumber <= filter.toBlock!);
+    }
+
+    if ("blockHash" in filter && filter.blockHash) {
+      logs = logs.filter((log) => log.blockHash <= filter.blockHash!);
+    }
```

Comment:
why this `if`s have this additional check (`"name" in filter`) and `address` and `topics` don't?